src/registration/metric/demons.h

```diff
@@ -47,6 +51,7 @@ namespace MR
           }
 
           ~Demons () {
+            std::lock_guard<std::mutex> lock (*mutex);
```

Comment:
Are you sure this is needed? I thought this was handled by [thread.h](https://github.com/MRtrix3/mrtrix3/blob/2bf78387e7b5dca6f2ea0adf0783b87b5f489bb1/lib/thread.h#L73)